flooddrake/timestepper.py

```diff
@@ -100,74 +117,69 @@ def __solver_variable(self, w):
         if self.mesh.geometric_dimension() == 1:
-            source = as_vector((-self.source_term * self.func(self.t), gravity * h * self.b_.dx(0)))
+            self.source = as_vector((-self.source_term*self.func(self.t), self.gravity * self.h * self.b_.dx(0)))
         if self.mesh.geometric_dimension() == 2:
-            source = as_vector((-self.source_term * self.func(self.t), gravity * h * self.b_.dx(0), gravity * h * self.b_.dx(1)))
+            self.source = as_vector((-self.source_term*self.func(self.t), self.gravity * self.h * self.b_.dx(0), self.gravity * self.h * self.b_.dx(1)))
 
         # solver - try to make this only once in the new version - by just assigning different values to all variable functions
         if self.mesh.geometric_dimension() == 2:
-            L = (dot(self.v, ((w_ - w) / self.dt)) * dx -
-                 dot(self.v.dx(0), F1) * dx -
-                 dot(self.v.dx(1), F2) * dx +
-                 (dot(self.v('-'), neg_flux) + dot(self.v('+'), pos_flux)) * dS +
-                 dot(self.v, boundary_flux) * ds +
-                 (dot(source, self.v)) * dx -
-                 (dot(self.v('-'), delta_minus) +
-                  dot(self.v('+'), delta_plus)) * dS)
+            self.L = (dot(self.v, ((self.w_ - self.w) / self.dt)) * dx -
+                      dot(self.v.dx(0), self.F1) * dx -
+                      dot(self.v.dx(1), self.F2) * dx +
+                      (dot(self.v('-'), self.NegFlux) + dot(self.v('+'), self.PosFlux)) * dS +
+                      dot(self.v, self.BoundaryFlux) * ds +
+                      (dot(self.source, self.v)) * dx -
+                      (dot(self.v('-'), self.delta_minus) +
+                      dot(self.v('+'), self.delta_plus)) * dS)
 
         if self.mesh.geometric_dimension() == 1:
-            L = (dot(self.v, ((w_ - w) / self.dt)) * dx -
-                 dot(self.v.dx(0), F) * dx +
-                 (dot(self.v('-'), neg_flux) + dot(self.v('+'), pos_flux)) * dS +
-                 dot(self.v, boundary_flux) * ds +
-                 (dot(source, self.v)) * dx -
-                 (dot(self.v('-'), delta_minus) +
-                  dot(self.v('+'), delta_plus)) * dS)
-
-        solve(L == 0, w_, nest=False,
-              solver_parameters={'ksp_type': 'preonly',
-                                 'pc_type': 'lu'})
+            self.L = (dot(self.v, ((self.w_ - self.w) / self.dt)) * dx -
+                      dot(self.v.dx(0), self.F) * dx +
+                      (dot(self.v('-'), self.NegFlux) + dot(self.v('+'), self.PosFlux)) * dS +
+                      dot(self.v, self.BoundaryFlux) * ds +
+                      (dot(self.source, self.v)) * dx -
+                      (dot(self.v('-'), self.delta_minus) +
+                      dot(self.v('+'), self.delta_plus)) * dS)
 
-        return w_
+        self.problem = NonlinearVariationalProblem(self.L, self.w_)
+        self.solver = NonlinearVariationalSolver(self.problem)
```

Comment:
You'll want to set appropriate solver parameters here:

```python
NonlinearVariationalSolver(self.problem, solver_parameters={...})
```

Note that `nest=False` lives with the `Problem`, not the `Solver`.